test/node/bson_binary_vector.spec.test.ts

```diff
@@ -87,21 +97,152 @@ const invalidTestExpectedError = new Map()
     'Invalid Vector: padding must be a value between 0 and 7'
   )
   .set('Negative padding PACKED_BIT', 'Invalid Vector: padding must be a value between 0 and 7')
-  // skipped
-  .set('Overflow Vector PACKED_BIT', false)
-  .set('Underflow Vector PACKED_BIT', false)
-  .set('Overflow Vector INT8', false)
-  .set('Underflow Vector INT8', false)
-  .set('INT8 with float inputs', false)
-  // duplicate test! but also skipped.
-  .set('Vector with float values PACKED_BIT', false)
-  .set('Vector with float values PACKED_BIT', false);
+  .set(
+    'Insufficient vector data FLOAT32',
+    'Invalid Vector: Float32 vector must contain a multiple of 4 bytes'
+  )
+  // These are not possible given the constraints of the input types allowed:
+  // our helpers will throw an "unsupported_error" for these
+  .set('Overflow Vector PACKED_BIT', 'unsupported_error')
+  .set('Underflow Vector PACKED_BIT', 'unsupported_error')
+  .set('Overflow Vector INT8', 'unsupported_error')
+  .set('Underflow Vector INT8', 'unsupported_error')
+  .set('INT8 with float inputs', 'unsupported_error')
+  .set('Vector with float values PACKED_BIT', 'unsupported_error');
+
+function catchError<T>(
+  fn: () => T
+): { status: 'returned'; result: T } | { status: 'thrown'; result: Error } {
+  try {
+    return { status: 'returned', result: fn() };
+  } catch (error) {
+    return { status: 'thrown', result: error };
+  }
+}
+
+function testVectorInvalidInputValues(test: VectorTest, expectedErrorMessage: string) {
+  describe('when creating a BSON Vector given invalid input values', () => {
+    const binaryCreation = catchError(make.bind(null, test.vector!, test.dtype_hex, test.padding));
+    const bsonBytesCreation =
+      binaryCreation.status !== 'thrown'
+        ? catchError(BSON.serialize.bind(null, { bin: binaryCreation.result }))
+        : undefined;
+    const ejsonStringCreation =
+      binaryCreation.status !== 'thrown'
+        ? catchError(BSON.EJSON.stringify.bind(null, { bin: binaryCreation.result }))
+        : undefined;
+
+    const binaryHelperValidations = [
+      'Padding specified with no vector data PACKED_BIT',
+      'Exceeding maximum padding PACKED_BIT',
+      'Negative padding PACKED_BIT',
+      ...Array.from(invalidTestExpectedError.entries())
+        .filter(([, v]) => v === 'unsupported_error')
+        .map(([k]) => k)
+    ];
+
+    const errorType = expectedErrorMessage === 'unsupported_error' ? Error : BSONError;
+    const errorName = expectedErrorMessage === 'unsupported_error' ? 'Error' : 'BSONError';
+
+    const check = outcome => {
+      expect(outcome).to.exist;
+      expect(outcome.status).to.equal('thrown');
+      expect(outcome.result).to.be.instanceOf(errorType);
+      expect(outcome.result)
+        .to.have.property('message')
+        .that.matches(new RegExp(expectedErrorMessage));
```

Comment:
```suggestion
      expect(outcome.result).to.match(new RegExp(expectedErrorMessage));
```

(optional) regular expressions convert their arguments to strings, so matching errors automatically asserts on the error message of the error